tests/failpoints/cases/test_conf_change.rs

```diff
@@ -204,3 +207,59 @@ fn test_stale_peer_cache() {
     fail::cfg("stale_peer_cache_2", "return").unwrap();
     cluster.must_put(b"k2", b"v2");
 }
+
+// The test is for this situation:
+// suppose there are 3 peers (1, 2, and 3) in a Raft group, and then
+// 1. propose to add peer 4 on the current leader 1;
+// 2. leader 1 appends entries to peer 3, and peer 3 applys them;
+// 3. the leadersip is transfered to peer 2, and peer 2 sends a snapshot to 3;
```

Comment:
If peer 2 sends a snapshot to 3, it means the snapshot has to contain configuration (1, 2, 3, 4). How can step 6 happen?